### ISSUE TITLE
Allow to edit schedules in the UI

### ISSUE THREAD

**Issue by adejanovski** [thelastpickle/cassandra-reaper#1006]
Schedules can't be updated, which means they have to be deleted and recreated to change a single setting.
We need to add that feature to improve the UX.

┆Issue is synchronized with this [Jira Task](https://k8ssandra.atlassian.net/browse/K8SSAND-247) by [Unito](https://www.unito.io)
┆Issue Number: K8SSAND-247

**Comment by jdonenine** [comment]
@adejanovski I wanted to tinker a bit in the Reaper UI and figured this issue might be a decent place to start, but I had a couple of questions...

1. Does the API to edit a schedule already exist?
2. What fields should be editable of an already existing schedule? (the answer to question 1 might answer this if the API already exists and only exposes updates to certain fields)

Thanks!

**Comment by adejanovski** [comment]
> 1. Does the API to edit a schedule already exist?

No it doesn't yet.

> 2. What fields should be editable of an already existing schedule? (the answer to question 1 might answer this if the API already exists and only exposes updates to certain fields)

Very good question! I think the safe (and easy path) is to allow changing everything that's specific to the schedule and avoid touching fields from the repair unit. That gives us the following fields:

- days_between (which will force recomputing next_activation)
- intensity
- owner
- repair_parallelism
- segment_count_per_node

The new API operation should only expose these fields and in the UI we could (possibly) simplify things by using a popup exposing these fields only. I foresee some tricky stuff if we try to use the current form (which has some shared bits with the repair run form).

**Comment by jdonenine** [comment]
I actually looked at the existing form and quickly decided I wasn't going to try to refactor that to make it support both use-cases.  My thought is generally that it should, but perhaps build out the new form and then make the existing form be able to leverage it and wrap it up.  The existing form does a lot of stuff it seems - a lot of stuff that I have no idea about :-)

**Comment by adejanovski** [comment]
I agree, there's a lot of logic in there that will make it hard to repurpose for schedule edition.

**Comment by jdonenine** [comment]
So let me know what you think about this...

My thinking here is to create a new form, that functions like the existing one, but that can be made a bit more flexible.  At the moment it's fairly rigid in what fields are available, but that could be enhanced later for reuse.

Adds an "Edit" launch for each schedule:

![image](https://user-images.githubusercontent.com/7901615/131436591-3c62f5ca-13a9-4011-a8e1-b78ac76a603f.png)

Launches a dialog with the form, only the fields that you mentioned before will be editable:

![image](https://user-images.githubusercontent.com/7901615/131436665-521a6039-c36b-49b9-b474-aae74e4dddb0.png)

![image](https://user-images.githubusercontent.com/7901615/131436698-031b1af0-e45f-4382-b221-f4c161277c7d.png)

It gives some room to enable editing of other fields later if that's something that we'd need, and should be pretty re-usable for the add workflow.

I've re-worked the visual layout slightly to be a bit more consistent across different devices too.

![image](https://user-images.githubusercontent.com/7901615/131436961-bd1a0304-fa74-49ad-a4f6-878c441dc14d.png)

**Comment by adejanovski** [comment]
Awesome stuff @jdonenine! 
One tiny suggestion: in the modal header, would it be possible to put the schedule id instead of `<cluster>/<keyspace>`?
While the id is not easy to read (it's a timeuuid), it'll be more accurate as a primary key since you could have several schedules for the same keyspace.
Other than that, it looks great to me 👍 
Thanks!

**Comment by jdonenine** [comment]
We can for sure include the ID, I went with the approach that I did though because the ID isn't referenced in the table at all, so I don't think that having the ID there really adds much association to things.  The other challenge there is the UUID is long and won't play well on smaller device resolutions. Although admittedly even a longer "name" will look poor there as well.

I'll play around with it and see how it looks, the other thing I'd suggest is that we could add the UUID inside the modal content and not in the header itself?
